xbmc/events/BaseEvent.cpp

```diff
@@ -20,6 +20,9 @@
 
 #include "BaseEvent.h"
 #include "guilib/LocalizeStrings.h"
+#include "utils/StringUtils.h"
+
+unsigned int CBaseEvent::s_nextIndex = 0;
```

Comment:
We have the millisecond one in CThread or you can use `std::chrono::steady_clock` which does nanosecond granularity on windows and Linux at least. I don't really like adding a static and state just to solve sorting. 

In missed that you use it in the format call.